### ISSUE TITLE
Add Profile-Guided Optimization (PGO) support to dart2native

### ISSUE THREAD

**Issue by zamazan4ik** [dart-lang/sdk#53855]
Hi!

Recently I started to evaluate the Profile-Guided Optimization (PGO) state across the whole ecosystem. I did a lot of PGO measurements - all the results are available [here](https://github.com/zamazan4ik/awesome-pgo). According to my experience, PGO helps with achieving better optimization levels in many real-world applications. That's why I think having PGO support in `dart2native` is a nice feature to have.

Multiple languages already support PGO like:

* C, C++, Fortran (in Clang, GCC, MSVC, ICC)
* Rust (Rustc)
* Go (was recently added in the 1.21 release)
* Java (via GraalVM)
* C#

Also, adding PGO to `dart2native` was already previously mentioned in https://github.com/dart-lang/sdk/issues/39367 - I just decided to create a dedicated issue for it.

**Comment by mkustermann** [comment]
> Also, adding PGO to dart2native was already previously mentioned in https://github.com/dart-lang/sdk/issues/39367 - I just decided to create a dedicated issue for it.

@zamazan4ik Thank you for filing the issue about PGO. Any reason we need 2 different github issues for this? Also notice that `dart2native` (aka `dart compile exe/aot-snapshot`) is not the only way to invoke our AOT compiler (e.g. `flutter build` also does invoke it). Maybe we can close this in favor of https://github.com/dart-lang/sdk/issues/53856?

**Comment by zamazan4ik** [comment]
> Any reason we need 2 different github issues for this?

Yeah, I will explain it. https://github.com/dart-lang/sdk/issues/53855 is about adding a possibility to compile Dart programs with PGO using `dart2native` (also, as you mentioned above `flutter build` and probably other AoT compiler invocations). https://github.com/dart-lang/sdk/issues/53856 is about compiling with PGO and BOLT the DartVM itself (which is written in C++). So these are two different issues.
